src/teacup.coffee

```diff
@@ -110,11 +147,19 @@ class Teacup
   renderContents: (contents, rest...) ->
     if not contents?
       return
-    else if typeof contents is 'function'
-      result = contents.apply @, rest
-      @text result if typeof result is 'string'
-    else
-      @text contents
+    @queue.push "contents: #{contents}", (done) =>
+      @queue.position = 0
+      if typeof contents is 'function'
+        if contents.length is 0 or contents.length is rest.length
+          result = contents.apply @, rest
+          @text result if typeof result is 'string'
+          done()
+        else if contents.length is 1
```

Comment:
this should probably be something more like `if contents.length is rest.length + 1`